File: AAC-Practice/app/src/main/java/com/bowoon/android/aac_practice/MemoDAO.java
package com.bowoon.android.aac_practice;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface MemoDAO {
    @Query("SELECT * FROM memo")
    LiveData<List<Memo>> loadAllMemo();

    @Insert
    void insert(Memo memo);

    @Delete
    void deleteMemo(Memo memo);
}

File: AACPracticeKotlin/app/src/main/java/com/bowoon/android/aac_practice_kotlin/MemoDAO.kt
package com.bowoon.android.aac_practice_kotlin

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query


@Dao
interface MemoDAO {
    @Query("SELECT * FROM memo")
    fun loadAllMemo(): LiveData<List<Memo>>

    @Insert
    fun insert(memo: Memo)

    @Delete
    fun deleteMemo(memo: Memo)
}